### DISCUSSION TITLE
How to properly handle api status 401 with useAsyncData or useFetch?

### DISCUSSION THREAD

**Discussion by wengys** [nuxt/nuxt#17309] [Questions] [answered]
On nuxt2, we load some data before page rendering, like below:
```ts
@Component({
  async asyncData({ $axios }) {
    let someData= await fetchSomeData($axios);
    return {
      someData,
    };
  },
})
export default class Index extends Vue {}
```
And we write a plugin, just simply check if some api returns status code 401, then redirect to login page:
```ts
export default function ({ $axios, redirect, $config }) {
    $axios.onError(error => {
        const code = parseInt(error.response && error.response.status)
        if(code === 401) {
            redirect($config.loginUrl)
        }
    })
}
```
That's it! Everything works fine. Good job nuxt2!
But when switching to nuxt3, we can't find anywhere globally hook api response as before. I searched for a while, but only found someone has mentioned similar problem with no clear answer what's the next move should I take.
Anyone can give suggestions?

**Comment by ytyng** [comment, marked as answer]
Get response status code from `useFetch` result:
https://github.com/nuxt/framework/discussions/3457#discussioncomment-2271626

```TypeScript
  const {data, error} = await useFetch<any>(...)
  if (error.value) {
    if (error.value.response.status === 401) {
      redirect($config.loginUrl)
    }
  }
```

**Reply by wengys** [reply]
Many thanks, my previous attempts struggling redirect inside useFetch, but yes I should not limit my mind, and a little modification since nuxt api changes
```ts
const {data, error} = await useFetch<any>(...)
if (error.value) {
  if(error.value instanceof FetchError) {
    if(error.value.response?.status === 401) {
      sendRedirect(useNuxtApp().ssrContext.event, $config.loginUrl, 302)
    }
  }
}
```
